jupyterlab_git/git.py

```diff
@@ -249,20 +255,16 @@ def __init__(self, contents_manager):
             }
 
         result = []
-        line_array = my_output.splitlines()
-        for line in line_array:
-            to1 = None
+        line_iterable = iter(strip_and_split(my_output))
+        for line in line_iterable:
             from_path = line[3:]
-            if line[0] == "R":
-                to0 = line[3:].split(" -> ")
-                to1 = to0[len(to0) - 1]
-            else:
-                to1 = line[3:]
-            if to1.startswith('"'):
-                to1 = to1[1:]
-            if to1.endswith('"'):
-                to1 = to1[:-1]
-            result.append({"x": line[0], "y": line[1], "to": to1, "from": from_path})
+            if line[0]=='R':
+                #If file was renamed then we need both this line
+                #and the next line, then we want to move onto the subsequent
+                #line. We can accomplish this by calling next on the iterable
+                from_path = next(line_iterable)
+            result.append({"x": line[0], "y": line[1], "to": line[3:], "from": from_path})
```

Comment:
```suggestion
            result.append({
                "x": line[0],
                "y": line[1],
                "to": line[3:],
                # if file was renamed, next line contains original path
                "from": next(line_iterable) if line[0]=='R' else line[3:]
            })       
```

This can be a bit clearer. Remove intermediate value `from_path` and use shorter/more succinct comment